src/Form/CalendarInstanceType.php

```diff
@@ -27,14 +27,21 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                 'help' => 'form.uri.help.caldav',
                 'required' => true,
             ])
-            ->add('public', ChoiceType::class, [
-                'label' => 'form.public',
-                'mapped' => false,
-                'disabled' => $options['shared'],
-                'help' => 'form.public.help.caldav',
-                'required' => true,
-                'choices' => ['yes' => true, 'no' => false],
-            ])
+            ->add(
+                'public',
+                $options['public_calendar_enabled'] ? ChoiceType::class : HiddenType::class,
+                $options['public_calendar_enabled'] ?  [
+                    'label' => 'form.public',
+                    'mapped' => false,
+                    'disabled' => $options['shared'],
+                    'help' => 'form.public.help.caldav',
+                    'required' => true,
+                    'choices' => ['yes' => true, 'no' => false],
+                ] : [
+                    'required' => true,
+                    'mapped' => false,
+                ]
+            )
```

Comment:
With a hidden form, you can still change the value if you want to. I'd rather remove the field (ie. not add it) altogether. Which means we need to test in the controller if 'public' is set or not in the form.